### ISSUE TITLE
happi fails to animate right in moving window

### ISSUE THREAD

**Issue by zubairy** [SmileiPIC/Smilei#282]
## Description

I open this issue since `happi` seems to fail to `animate` in the right way in moving window again. The previous related issue is #69 .

The following figure show a LWFA result (Smilei Version: bv4.4-706-gb5c12a5a-bmaster) with down-ramp injection at ~300 \lambda. The upper row using `happi` in 4bfc3886 shows the expected result, while in the lower (`happi` in b5c12a5a, HEAD of current master branch), both the x-axis and the data is incorrect (seems like the x-axis failed to catch up with the data). 

`plot` in both `happi` version give correct result. Could you please kindly check were there some other bugs being introduced into the code since issue #69 ? Thanks.

![image](https://user-images.githubusercontent.com/12051096/89154003-ab113b80-d598-11ea-93b6-683649a6f369.png)

**Comment by zubairy** [comment]
Hi, I add the following steps if it can help to reproduce this bug quickly.

## Steps to reproduce the problem
1. use `benchmarks/tst2d_04_laser_wake.py` as input file;
2. Post-Processing code:
    ```python
    import happi
    from math import pi
    Wr = 2.0*pi*299792458/0.8e-6
    S=happi.Open('./',reference_angular_frequency_SI=Wr)
    Ex=S.Field(diagNumber=0,field='Ex',moving=True,units=['um','V/m','fs'],cmap='seismic')
    Ex.animate(saveAs='Ex_')
    ```
3. From timestep=900, the bug will appear.

## Parameters

- Smilei version: bv4.4-706-gb5c12a5-bmaster
- Input file: benchmarks/tst2d_04_laser_wake.py
- Computing resources
  - type of computer: cluster
  - OS: CentOS Linux 7
  - C++ compiler (to get some information `g++ --version`): g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)
  - MPI version (`mpic++ --version`, `mpic++ -show`): intel/2019/mpirun
  - HDF5 version (`h5pcc --version`): icc (ICC) 19.0.4.243 20190416
  - Python version: python 3.7.1

**Comment by mccoys** [comment]
Thank you for the details. We will look at this problem asap, but nobody is available this week.

**Comment by mccoys** [comment]
Could you try the last commit 360fc39 ?

**Comment by zubairy** [comment]
Thanks @mccoys , commit 360fc39 resoved this issue.

**Comment by zubairy** [comment]
Sorry to reopen, but if processing a single `timestep`, animate seems can not save pictures. For example (still using `tst2d_04_laser_wake.py`):

`S.Field(diagNumber=0,field='Ex',moving=True,units=['um','V/m','fs'],timesteps=[300],cmap='seismic').animate(saveAs='Ex_')`, 

no picture saved.

`timesteps=[201,300]`, no picture neither.

`timesteps=[200,300]`, saved `Ex_300.png` but no `Ex_200.png`.

`timesteps=None`, saved all other pictures without 'Ex_000.png'.

Can you reproduce this ?

**Comment by mccoys** [comment]
Thanks for spotting this bug. Fixed in 974792f

**Comment by zubairy** [comment]
Thank you for this quick fix.
